salt/state.py

```diff
@@ -702,6 +702,7 @@ def __init__(
         self.instance_id = str(id(self))
         self.inject_globals = {}
         self.mocked = mocked
+        self.guards = salt.loader.guards(self.opts)
```

Comment:
since load_modules has been called we should be able to load in the execution modules as `__salt__` as well. We should also consider making the guards have access to all of the dunders that the state modules have access to, like `__states__`